addon/-private/fields/attr.ts

```diff
@@ -1,24 +1,62 @@
-import { computed } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 import { Dict } from '@orbit/utils';
 
-export default function (type: string, options: Dict<unknown> = {}) {
-  options.type = type;
+import Model from '../model';
+
+export default function attr(type: string, options: Dict<unknown> = {}) {
+  function trackedAttr(target: any, key: string, desc: PropertyDescriptor) {
+    let trackedDesc = tracked(target, key, desc);
+    let { get: originalGet, set: originalSet } = trackedDesc;
+
+    let defaultAssigned = new WeakSet();
+
+    function setDefaultValue(record: Model) {
+      let value = record.getAttribute(key);
+      setValue(record, value);
+    }
+
+    function setValue(record: any, value: any) {
+      defaultAssigned.add(record);
+      return originalSet!.call(record, value);
+    }
+
+    function get(this: Model) {
```

Comment:
```suggestion
    function get(record: Model) {
```

Wouldn't it be better to keep the argument names consistent?

Reply:
`this` is not argument name but special TS thing to provide `this` type

Reply:
In this case `get` takes 0 arguments but it is executed on the object of type `Model`

Reply:
Ah, thanks for the clarification! This made no sense to me from a JS perspective, but I can now see what TS is doing.